server/sonar-server/src/main/java/org/sonar/server/computation/issue/DebtCalculator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.issue;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Strings;
+import javax.annotation.CheckForNull;
+import org.sonar.api.server.debt.DebtRemediationFunction;
+import org.sonar.api.server.debt.DebtRemediationFunction.Type;
+import org.sonar.api.utils.Duration;
+import org.sonar.api.utils.Durations;
+import org.sonar.core.issue.DefaultIssue;
+
+public class DebtCalculator {
+
+  private final RuleRepository ruleRepository;
+  private final Durations durations;
+
+  public DebtCalculator(RuleRepository ruleRepository, Durations durations) {
+    this.ruleRepository = ruleRepository;
+    this.durations = durations;
+  }
+
+  @CheckForNull
+  public Duration calculate(DefaultIssue issue) {
+    Rule rule = ruleRepository.getByKey(issue.ruleKey());
+    DebtRemediationFunction fn = rule.getRemediationFunction();
+    if (fn != null) {
+      verifyEffortToFix(issue, fn);
+
+      Duration debt = Duration.create(0);
+      if (fn.type().usesCoefficient() && !Strings.isNullOrEmpty(fn.coefficient())) {
+        int effortToFixValue = Objects.firstNonNull(issue.effortToFix(), 1).intValue();
+        // TODO convert to Duration directly in Rule#remediationFunction -> better performance + error handling
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)